packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts

```diff
@@ -0,0 +1,174 @@
+  }
+
+  field = (
+    parent: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode,
+    definition: FieldDefinitionNode,
+    directive: DirectiveNode,
+    acc: TransformerSchemaVisitStepContextProvider,
+  ): void => {
+    const directiveWrapped = new DirectiveWrapper(directive);
+    const args = directiveWrapped.getArguments({
+      resolverTypeName: parent.name.value,
+      resolverFieldName: definition.name.value,
+    } as FunctionDirectiveConfiguration);
+    let resolver = this.resolverGroups.get(definition);
+
+    if (resolver === undefined) {
+      resolver = [];
+      this.resolverGroups.set(definition, resolver);
+    }
+
+    resolver.push(args);
+  };
+
+  generateResolvers = (context: TransformerContextProvider): void => {
+    if (this.resolverGroups.size === 0) {
+      return;
+    }
+
+    const stack: cdk.Stack = context.stackManager.createStack(FUNCTION_DIRECTIVE_STACK);
+    const createdResources = new Map<string, any>();
+    const env = context.stackManager.getParameter(ResourceConstants.PARAMETERS.Env) as cdk.CfnParameter;
+
+    stack.templateOptions.templateFormatVersion = '2010-09-09';
+    stack.templateOptions.description = 'An auto-generated nested stack for the @function directive.';
+
+    new cdk.CfnCondition(stack, ResourceConstants.CONDITIONS.HasEnvironmentParameter, {
+      expression: cdk.Fn.conditionNot(cdk.Fn.conditionEquals(env, ResourceConstants.NONE)),
+    });
+
+    this.resolverGroups.forEach((resolverFns, fieldDefinition) => {
+      resolverFns.forEach(config => {
+        // Create data sources that register Lambdas and IAM roles.
+        const dataSourceId = FunctionResourceIDs.FunctionDataSourceID(config.name, config.region);
+
+        if (!createdResources.has(dataSourceId)) {
+          const dataSource = context.api.addLambdaDataSource(
+            dataSourceId,
+            lambda.Function.fromFunctionAttributes(stack, `${dataSourceId}Function`, {
+              functionArn: lambdaArnResource(env, config.name, config.region),
+            }),
+            {},
+            stack,
+          );
+          createdResources.set(dataSourceId, dataSource);
+        }
+
+        // Create AppSync functions.
+        const functionId = FunctionResourceIDs.FunctionAppSyncFunctionConfigurationID(config.name, config.region);
+        let func = createdResources.get(functionId);
+
+        if (func === undefined) {
+          func = context.api.addAppSyncFunction(
+            functionId,
+            MappingTemplate.s3MappingTemplateFromString(
+              printBlock(`Invoke AWS Lambda data source: ${dataSourceId}`)(
+                obj({
+                  version: str('2018-05-29'),
+                  operation: str('Invoke'),
+                  payload: obj({
+                    // TODO(cjihrig): The typeName and fieldName below seem potentially wrong (#6419)
+                    typeName: str('$ctx.stash.get("typeName")'),
```

Comment:
this could be fixed by chainging this from `str` to `ref`
```suggestion
                    typeName: ref('ctx.stash.get("typeName")'),
```